cupy/creation/ranges.py

```diff
@@ -126,7 +126,66 @@ def logspace(start, stop, num=50, endpoint=True, base=10.0, dtype=None):
     return core.power(base, y).astype(dtype)
 
 
-# TODO(okuta): Implement meshgrid
+def meshgrid(*xi, **kwargs):
+    """Return coordinate matrices from coordinate vectors.
+
+    Given one-dimensional coordinate arrays x1, x2, ..., xn, this function
+    makes N-D grids.
+
+    For ndarrays x1, x2, ..., xn with lengths ``Ni = len(xi)``, return
+    ``(N1, N2, N3, ..., Nn)`` shaped arrays if indexing='ij' or
+    ``(N2, N1, N3, ..., Nn)`` shaped arrays if indexing='xy'.
+
+    Unlike NumPy, CuPy currently only supports 1-D arrays as inputs.
+    Also, CuPy does not support ``sparse`` option yet.
+
+    Args:
+        xi (tuple of ndarrays): 1-D arrays representing the coordinates
+            of a grid.
+        indexing ({'xy', 'ij'}, optional): Cartesian ('xy', default) or
+            matrix ('ij') indexing of output.
+        copy (bool, optional): If ``False``, a view
+            into the original arrays are returned. Default is True.
+
+    Returns:
+        list of cupy.ndarray
+
+    .. seealso:: :func:`numpy.meshgrid`
+
+    """
+
+    indexing = kwargs.pop('indexing', 'xy')
+    copy = bool(kwargs.pop('copy', True))
+    if indexing not in ['xy', 'ij']:
+        raise ValueError('Valid values for `indexing` are \'xy\' and \'ij\'.')
+
```

Comment:
This may be not so important, but how about check if `kwargs` contains unnecessary arguments as is in numpy implementation? https://github.com/numpy/numpy/blob/v1.12.0/numpy/lib/function_base.py#L4504